test/unit/transformRemappings.spec.ts

```diff
@@ -5,35 +5,137 @@ import mock from 'mock-fs';
 describe('transformRemappings', () => {
   before(() => {
     mock({
-      'remappings.txt': 'interfaces=../../interfaces',
+      'remappings.txt': '',
     });
   });
-
   after(() => {
     mock.restore();
   });
+  describe('with remapping for local file only', function () {
+    before(function () {
+      mock({
+        'remappings.txt': 'interfaces=../../interfaces',
+      });
+    });
+    it('should replace path as instructed by remapping with all import syntaxes', function () {
+      const fileContent = `
+                                  import '../../../interfaces/ITest.sol';
```

Comment:
I might be wrong here, but I don't think remappings work this way. If your remapping key is `interfaces`, `../../../interfaces` should not be affected, only when it starts with `interfaces/`

Reply:
:eyes: will check

Reply:
forge does a weird thing where it complains in the logs it cant find the files, as if it only used a remapping when the key is at the start of the imported path as you describe, but then replaces the key anywhere and manages to build the project. as a PoC, I:

- ran `forge init`
- in that repo, added `wherecodeis/=src/`
- then edited the `Counter` import to `import {Counter} from "../wherecodeis/Counter.sol";`

and got this log:
```
[N] ~/p/remappingexample (master){2}
> forge test
[⠊] Compiling...
[⠒] Unable to resolve imports:
      "../wherecodeis/Counter.sol" in "/home/user/playground/remappingexample/test/Counter.t.sol"
with remappings:
      wherecodeis/=/home/user/playground/remappingexample/src/
      forge-std/=/home/user/playground/remappingexample/lib/forge-std/src/
[⠰] Compiling 25 files with Solc 0.8.26
[⠔] Solc 0.8.26 finished in 1.20s
Compiler run successful!

Ran 2 tests for test/Counter.t.sol:CounterTest
[PASS] testFuzz_SetNumber(uint256) (runs: 256, μ: 31210, ~: 31288)
[PASS] test_Increment() (gas: 31303)
Suite result: ok. 2 passed; 0 failed; 0 skipped; finished in 7.01ms (6.73ms CPU time)

Ran 1 test suite in 8.85ms (7.01ms CPU time): 2 tests passed, 0 failed, 0 skipped (2 total tests)
```

Reply:
as forge considers the remapping valid, I'd say to keep the current behaviour, however I'm open to structuring the tests in a different way to make this edge case more obvious

Reply:
> structuring the tests in a different way to make this edge case more obvious

Please 🙏🏻

Reply:
5f90b8c